core/templates/components/forms/custom-forms-directives/image-uploader.component.ts

```diff
@@ -0,0 +1,151 @@
+import {Component, Input} from '@angular/core';
+import {downgradeComponent} from '@angular/upgrade/static';
+import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import {UrlInterpolationService} from 'domain/utilities/url-interpolation.service';
+import {ContextService} from 'services/context.service';
+import {ImageUploadHelperService} from 'services/image-upload-helper.service';
+import {ImageUploaderModalComponent} from './image-uploader-modal.component';
+
+export interface ImageUploaderParameters {
+  disabled: boolean;
+  maxImageSizeInKB: number;
+  imageName: string;
+  orientation: string;
+  bgColor: string;
+  allowedBgColors: string[];
+  allowedImageFormats: string[];
+  aspectRatio: string;
+  filename?: string;
+  previewTitle?: string;
+  previewDescription?: string;
+  previewDescriptionBgColor?: string;
+  previewFooter?: string;
+  previewImageUrl?: string;
+}
+
+@Component({
+  selector: 'oppia-image-uploader',
+  templateUrl: './image-uploader.component.html',
+})
+export class ImageUploaderComponent implements OnInit {
+  @Output() updateBgColor: EventEmitter<string> = new EventEmitter();
+  @Output() updateFilename: EventEmitter<string> = new EventEmitter();
+  @Output() imageSave: EventEmitter<Blob> = new EventEmitter();
+
+  @Input() imageUploaderParameters!: ImageUploaderParameters;
+
+  dimensions!: {height: number; width: number};
+  editableImageDataUrl!: string;
+  hidePlaceholder: boolean = false;
+  imageBgColor!: string;
+  placeholderImageUrl!: string;
+
+  constructor(
+    private imageUploadHelperService: ImageUploadHelperService,
+    private contextService: ContextService,
+    private ngbModal: NgbModal,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  ngOnInit(): void {
+    this.placeholderImageUrl = this.urlInterpolationService.getStaticImageUrl(
+      this.isImageInPortraitMode()
+        ? '/icons/story-image-icon.png'
+        : '/icons/story-image-icon-landscape.png'
+    );
+
+    if (this.imageUploaderParameters.filename) {
+      const entityType = this.contextService.getEntityType();
+      if (entityType === undefined) {
+        throw new Error('No image present for preview');
+      }
+      this.editableImageDataUrl =
+        this.imageUploadHelperService.getTrustedResourceUrlForThumbnailFilename(
+          this.imageUploaderParameters.filename,
+          entityType,
+          this.contextService.getEntityId()
+        );
+      this.imageBgColor = this.imageUploaderParameters.bgColor;
+      this.imageUploaderParameters.previewImageUrl = this.editableImageDataUrl;
+      this.hidePlaceholder = true;
+    }
+  }
+
+  isImageInPortraitMode(): boolean {
+    return this.imageUploaderParameters.orientation === 'portrait';
+  }
+
+  showImageUploaderModal(): void {
+    if (this.imageUploaderParameters.disabled) {
+      return;
+    }
+
+    this.dimensions = {
+      height: 0,
+      width: 0,
+    };
```

Comment:
Is this the only place you set dimensions? That's suspicious

Also, if the only place these dimensions are used is the filename generation, then there's no need to make it a state, just create those dimensions right where you need them

Reply:
Removed it.